.github/workflows/nix-managed-lints.yml

```diff
@@ -0,0 +1,51 @@
+# ============================================================================ #
+#
+# Run `pre-commit` over the codebase.
+#
+# ---------------------------------------------------------------------------- #
+
+name: "Project Lints"
+on:
+  pull_request:
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.head_ref || github.sha }}"
+  cancel-in-progress: true
+
+jobs:
+  nix-git-hooks:
+    name: "Nix Git Hooks"
+    runs-on: "ubuntu-latest"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v4"
+
+      - name: "Setup Nix"
+        uses: "./.github/actions/common-setup"
+        with:
+          GITHUB_ACCESS_TOKEN: "${{ secrets.NIX_GIT_TOKEN }}"
+          SUBSTITUTER: "${{ vars.FLOX_CACHE_PUBLIC_BUCKET }}"
+          SUBSTITUTER_KEY: "${{ secrets.FLOX_CACHE_PUBLIC_NIX_SECRET_KEY }}"
+          AWS_ACCESS_KEY_ID: "${{ secrets.FLOX_CACHE_PUBLIC_AWS_ACCESS_KEY_ID }}"
+          AWS_SECRET_ACCESS_KEY: "${{ secrets.FLOX_CACHE_PUBLIC_AWS_SECRET_ACCESS_KEY }}"
+          SSH_KEY: "${{ secrets.FLOXBOT_SSH_KEY }}"
```

Comment:
(curious, non-blocking) What are these needed for?

Reply:
Access to the nix binary cache that we host on our s3
I dont think they are needed here, but i also didnt check whether they are truly optional in the common-setup action